vm/systemSmartContracts/staking.go

```diff
@@ -614,6 +628,47 @@ func (r *stakingSC) unStake(args *vmcommon.ContractCallInput) vmcommon.ReturnCod
 	return vmcommon.Ok
 }
 
+func (r *stakingSC) moveFirstFromWaitingToStaked(blsKey []byte) error {
+	waitingList, err := r.getWaitingListHead()
+	if err != nil {
+		return err
+	}
+	if waitingList.Length == 0 {
+		return nil
+	}
+
+	elementInList, err := r.getWaitingListElement(waitingList.FirstKey)
+	if err != nil {
+		return err
+	}
+	if bytes.Equal(blsKey, elementInList.BLSPublicKey) {
```

Comment:
If the node which is doing unStake is the firstKey in waiting list why we do return here instead of calling the removeFromWaitingList as it is and afterward executing the following mechanism with the elementInList.NextKey if they exist?

Reply:
done. refactored